src/Raven.Client/Json/TypeConverter.cs

```diff
@@ -63,13 +62,17 @@ public static object ToBlittableSupportedType(object value, DocumentConventions
                 return @object;
             }
 
-            if (value is IEnumerable enumerable)
+            if (value is IEnumerable)
```

Comment:
there might be some perf benefit checking for IList **if** it's common input, now enumerator is always allocated vs direct indexing using old-school for-loop

Reply:
I am not sure it will work, 
the typeof value received here is object, so in both cases (IList or IEnuramble) we will have to cast it. Because we want to just enumerate on collection (we dont need the indexes that are present in lists) I dont see a point to make a if-case for IList type.

Reply:
Fair enough, I merely wanted to point out there can also be a performance difference which way you do it. [Here's some numbers](https://blog.lahma.net/2018/11/optimizing-jint-part-4-interfaces-are.html) for iterating the same List with different casts. So not entirely related whether you need the indexes used when iterating.

But as said, highly usage scenario dependent (if most are not lists, no point of having extra complexity). With IList there would still be interface dispatch slowing things down, but "a lot" faster than iterating IEnumerable.

Reply:
What do we get here depends on an index definition - so can't assume it will be IList mostly